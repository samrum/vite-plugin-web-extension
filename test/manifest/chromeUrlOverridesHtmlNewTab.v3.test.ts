import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("chromeUrlOverridesHtml");

runManifestV3Test("chromeUrlOverridesHtmlNewTab", () => ({
  chrome_url_overrides: {
    newtab: `${resourceDir}/newtab.html`,
  },
}));
