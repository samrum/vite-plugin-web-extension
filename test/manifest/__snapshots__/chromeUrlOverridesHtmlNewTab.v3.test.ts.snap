// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`chromeUrlOverridesHtmlNewTab - Manifest V3 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
log(\\"chromeUrlOverridesHtml\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/chromeUrlOverridesHtml/newtab.html",
    "fileName": "assets/test/manifest/resources/chromeUrlOverridesHtml/newtab.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/chromeUrlOverridesHtml/chromeUrlOverridesHtml.js": {
        "code": "log(\\"chromeUrlOverridesHtml\\");",
        "originalLength": 67,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 30,
      },
      "vite-plugin-web-extension/test/manifest/resources/chromeUrlOverridesHtml/newtab.html": {
        "code": "",
        "originalLength": 241,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "test/manifest/resources/chromeUrlOverridesHtml/newtab",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 3,
  \\"chrome_url_overrides\\": {
    \\"newtab\\": \\"test/manifest/resources/chromeUrlOverridesHtml/newtab.html\\"
  }
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/chromeUrlOverridesHtml/newtab.html",
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />

    <script type=\\"module\\" src=\\"http://example.com/httpModuleScript.js\\"></script>
    <script type=\\"module\\" crossorigin src=\\"/assets/test/manifest/resources/chromeUrlOverridesHtml/newtab.js\\"></script>
  </head>
</html>
",
    "type": "asset",
  },
]
`;
